core/domain/exp_fetchers.py

```diff
@@ -194,29 +212,61 @@ def get_exploration_from_model(exploration_model, run_conversion=True):
         last_updated=exploration_model.last_updated)
 
 
+@overload
 def get_exploration_summary_by_id(
-    exploration_id: str
-) -> exp_domain.ExplorationSummary:
+    exploration_id: str,
+) -> exp_domain.ExplorationSummary: ...
+
+
+@overload
+def get_exploration_summary_by_id(
+    exploration_id: str,
+    strict: Literal[True]
+) -> exp_domain.ExplorationSummary: ...
+
+
+@overload
+def get_exploration_summary_by_id(
+    exploration_id: str,
+    strict: Literal[False]
+) -> Optional[exp_domain.ExplorationSummary]: ...
+
+
+@overload
+def get_exploration_summary_by_id(
+    exploration_id: str,
+    strict: bool = False
+) -> Optional[exp_domain.ExplorationSummary]: ...
+
+
+def get_exploration_summary_by_id(
+    exploration_id: str,
+    strict: bool = True
+) -> Optional[exp_domain.ExplorationSummary]:
```

Comment:
Why strict is introduced!

Reply:
In some places in the codebase, we are not handling the case of `None` for this function's return value, for eg:
https://github.com/oppia/oppia/blob/29f4595bd58acfed0f1988bc16abeba78cc0d7e4/core/controllers/editor_test.py#L1928
https://github.com/oppia/oppia/blob/29f4595bd58acfed0f1988bc16abeba78cc0d7e4/core/domain/exp_services.py#L1317